src/common/modules/AutocorrectHandler.js

```diff
@@ -0,0 +1,198 @@
+"use strict";
+
+import * as AddonSettings from "/common/modules/AddonSettings/AddonSettings.js";
+import * as BrowserCommunication from "/common/modules/BrowserCommunication/BrowserCommunication.js";
+
+import { COMMUNICATION_MESSAGE_TYPE } from "/common/modules/data/BrowserCommunicationTypes.js";
+import * as symbols from "/common/modules/data/Symbols.js";
+
+const settings = {
+    autocorrectEmojis:  null,
+    quotes:  null,
+    fracts:  null,
+};
+
+let autocorrections = {};
+
+// Longest autocorrection
+let longest = 0;
+
+let symbolpatterns = [];
+// Do not autocorrect for these patterns
+let apatterns = [];
```

Comment:
based on the comment then maybe `antipatterns`? (IMHO a better, more self-epxlaining variable name.)